test/service/test_request_validation.py

```diff
@@ -87,11 +87,11 @@ def tearDown(self):
         if self.server_thread.is_alive():
             self.fail('Thread is still alive after joining')
 
-    def test_hello_world(self):
+    def test_health(self):
         url = self.base_url
         response = requests.get(url)
         self.assertEqual(response.status_code, 200, response.json())
-        self.assertEqual({"Hello": "World!"}, response.json())
+        self.assertEqual({"Status": "UP"}, response.json())
```

Comment:
I don't see how this test can pass because the response will have additional elements in it.

Reply:
Does the test pass for you? Did you run it?

Reply:
@hannes-ucsc I forgot to update it. Anyway I'm not able to run tests. I've tried running them directly in PyCharm and also `make test`. Both options fails before getting to testing. If none of the options I've mentioned is default, we should have it covered in readme.

```
Testing started at 8:02 PM ...
/home/milan/.virtualenvs/.azul/bin/python /home/milan/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/182.3684.100/helpers/pycharm/_jb_pytest_runner.py --target test_request_validation.py::FacetNameValidationTest.test_health
Launching pytest with arguments test_request_validation.py::FacetNameValidationTest::test_health in /home/milan/Projects/human-cell-atlas/azul/test/service

============================= test session starts ==============================
platform linux -- Python 3.7.0, pytest-3.2.3, py-1.5.4, pluggy-0.4.0
rootdir: /home/milan/Projects/human-cell-atlas/azul/test/service, inifile:
collected 1 item
test_request_validation.py EERROR:shared:Docker container took more than a minute to set up

test setup failed
cls = <class 'service.test_request_validation.FacetNameValidationTest'>

    @classmethod
    def setUpClass(cls):
>       super().setUpClass()

test_request_validation.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
__init__.py:17: in setUpClass
    super().setUpClass()
../shared.py:39: in setUpClass
    cls._wait_for_es()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'service.test_request_validation.FacetNameValidationTest'>

    @classmethod
    def _wait_for_es(cls):
        patched_log_level = logging.WARNING if logger.getEffectiveLevel() <= logging.DEBUG else logging.ERROR
        start_time = time.time()
        with patch.object(logging.getLogger('elasticsearch'), 'level', new=patched_log_level):
            while not cls.es_client.ping():
                if time.time() - start_time > 60:
                    logger.error('Docker container took more than a minute to set up')
>                   raise AssertionError
E                   AssertionError

../shared.py:49: AssertionError


==================================== ERRORS ====================================
____________ ERROR at setup of FacetNameValidationTest.test_health _____________

cls = <class 'service.test_request_validation.FacetNameValidationTest'>

    @classmethod
    def setUpClass(cls):
>       super().setUpClass()

test_request_validation.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
__init__.py:17: in setUpClass
    super().setUpClass()
../shared.py:39: in setUpClass
    cls._wait_for_es()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'service.test_request_validation.FacetNameValidationTest'>

    @classmethod
    def _wait_for_es(cls):
        patched_log_level = logging.WARNING if logger.getEffectiveLevel() <= logging.DEBUG else logging.ERROR
        start_time = time.time()
        with patch.object(logging.getLogger('elasticsearch'), 'level', new=patched_log_level):
            while not cls.es_client.ping():
                if time.time() - start_time > 60:
                    logger.error('Docker container took more than a minute to set up')
>                   raise AssertionError
E                   AssertionError

../shared.py:49: AssertionError
---------------------------- Captured stderr setup -----------------------------
ERROR:shared:Docker container took more than a minute to set up
========================== 1 error in 174.52 seconds ===========================
Process finished with exit code 0
```

Reply:
When you run `docker ps` while the test is spinning up, do you see a container for elasticsearch? If so, try attaching to it and see if it prints anything interesting. Also `docker logs` might give some insight.

Reply:
@GPelayo is currently working on making `make test` functional again. The currently prescribed method for running tests is PyCharm with envhook installed in the virtualenv. @achave11 is adding docs about that to the README.

I do remember us installing envhook into your virtualenv while we were troubleshooting the project setup on your system but I don't see its effect in your log above. Without envhook installed, tests won't work either but they would fail at a later stage.

@alex-hancock does the error above look familiar?

Reply:
Not at all. @mpanik do you have Docker installed on your machine? If you haven't installed it from [here](https://www.docker.com/community-edition), you might do that and try it again. I don't know why else it would take so long to set up.